### PR TITLE
OBSDOCS-1146: Documentation for the group-by-attribute processor

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is for https://github.com/openshift/openshift-docs/tree/dtotel-docs-3.3, not `main`

Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16, 4.17

Issue:

https://issues.redhat.com/browse/OBSDOCS-1146

Link to docs preview:

https://78990--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-processors.html#groupbyattrsprocessor-processor_otel-collector-processors

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by IshwarKanse — APPROVED** [review]

File: observability/otel/otel-collector/otel-collector-processors.adoc [hunk 1]
```diff
@@ -430,3 +430,26 @@ config: |
 <2> Defines a value provided in the `metrics` field as a `strict` exact match or `regexp` regular expression.
 <3> Lists the metric names, which are exact matches or matches for regular expressions, of the metrics to be converted to delta metrics. If a metric matches both the `include` and `exclude` filters, the `exclude` filter takes precedence.
 <4> Optional: Configures which metrics to exclude. When omitted, no metrics are excluded from conversion to delta metrics.
+
+[id="groupbyattrsprocessor-processor_{context}"]
+== Group-by-Attributes Processor
+
+This processor groups all spans, log records, and metric datapoints that share the same attributes by reassigning them to a Resource that matches those attributes.
```
Comment (jneczypor) [inline observability/otel/otel-collector/otel-collector-processors.adoc:437]:
```suggestion
The Group-by-Attributes Processor groups all spans, log records, and metric datapoints that share the same attributes by reassigning them to a Resource that matches those attributes.
```
All the other processors seem to start this way (barring the exception of Cumulative to Delta Processor).

File: observability/otel/otel-collector/otel-collector-processors.adoc [hunk 2]
```diff
@@ -430,3 +430,26 @@ config: |
...
+
+:FeatureName: The Group-by-Attributes Processor
+include::snippets/technology-preview.adoc[]
+
+Configuring this processor involves at minimum specifying an array of attribute keys to be used to group spans, log records, or metric datapoints together, as in the following example:
```
Comment (jneczypor) [inline observability/otel/otel-collector/otel-collector-processors.adoc:442]:
```suggestion
At minimum, configuring this processor involves specifying an array of attribute keys to be used to group spans, log records, or metric datapoints together, as in the following example:
```

File: observability/otel/otel-collector/otel-collector-processors.adoc [hunk 3]
```diff
@@ -430,3 +430,26 @@ config: |
...
+# ...
+processors:
+  groupbyattrs:
+    keys: # <1>
+      - <key1> # <2>
+      - <key2>
+# ...
+----
+<1> Specifies attribute keys to group by.
+<2> If a processed span, log record, or metric datapoint contains at least one of the specified attribute keys, it is reassigned to a Resource that shares the same attribute values; and if no such Resource exists, a new one is created. Also, if none of the specified attribute keys is present in the processed span, log record, or metric datapoint, then it remains associated with its current Resource. Multiple instances of the same Resource are consolidated.
```
Comment (jneczypor) [inline observability/otel/otel-collector/otel-collector-processors.adoc:455]:
```suggestion
<2> If a processed span, log record, or metric datapoint contains at least one of the specified attribute keys, it is reassigned to a Resource that shares the same attribute values. If no such Resource exists, a new one is created. If none of the specified attribute keys is present in the processed span, log record, or metric datapoint, then it remains associated with its current Resource. Multiple instances of the same Resource are consolidated.
```

**Review by jneczypor — COMMENTED** [review]
Just a few notes. Otherwise, LGTM.
/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: observability/otel/otel-collector/otel-collector-processors.adoc (hunk 1 above)
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-processors.adoc:437]: Updated. Thank you.

File: observability/otel/otel-collector/otel-collector-processors.adoc (hunk 2 above)
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-processors.adoc:442]: Updated. Thank you.

File: observability/otel/otel-collector/otel-collector-processors.adoc (hunk 3 above)
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-processors.adoc:455]: Thank you. I kept the first sentence as is (for better readability despite extra length) and updated the second sentence as you suggested.

**Pull request merged**